### PR TITLE
Feat/searchform

### PR DESCRIPTION
**Pull request by danieljhegeman**
Hey @Andyderson ! If you could give my code a look-see when you get a chance, that'd be great. I know there's a lot here--I really should've broken this into two or three separate PR's--but if you could focus on the query logic and test that I wrote that'd be great. Thanks!

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -14,12 +15,20 @@ app.use(express.static(path.join(__dirname, '../public')));
...
+      ${req.params.field} asc`)
+    .then(result => res.send(result))
+    .catch(err => res.send(err));
+});
+
 app.use('/restaurants', function(req, res) {
   //const fields = ['name', 'cuisine', 'neighborhood', 'price', 'vegetarian', 'byob'];
   var searchParams = req.body;
   var query = [`select iterator, name, cuisine, 
     neighborhood, price, vegetarian,
```
Comment (Uhan19) [inline server/index.js]: Should have an event listener for vegetarian that deletes restaurant from DB when true. Wait a minute...

File: client/components/search.jsx [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 
 class Search extends React.Component {
 
   constructor(props) {
     super(props);
     this.executeSearch = this.props.executeSearch;
     this.state = {};
-    this.stringFields = ['Name', 'Neighborhood'];
-    this.cuisines = ['American', 'Chinese', 'Filipino', 'French', 'Italian', 'Japanese', 'Korean', 'Mexican']; 
     this.checkboxFields = ['Vegetarian', 'BYOB'];
```
Comment (Andyderson) [inline client/components/search.jsx]: Should have an event listener for vegetarian that deletes restaurant from DB when true. Wait a minute...

File: client/components/search.jsx [hunk 3]
```diff
@@ -38,22 +37,11 @@ class Search extends React.Component {
...
-            <label>Cuisine:</label>
-            <select id="cuisine" className="query-field" >
-              <option value="" ></option>
-              {this.cuisines.map(cuisine => {
-                  return <option value={cuisine}>{cuisine}</option>
-                })
-              }
-            </select>
-            <div>
+            <Field type='text' name="Name" />
```
Comment (stephaniesara) [inline client/components/search.jsx]: Good refactor!

File: client/components/sidebar.jsx [hunk 4]
```diff
@@ -14,8 +14,23 @@ class SideBar extends React.Component {
...
       showSearch: false,
       showSearchResults: false,
-      searchButtonVal: 'Search restaurants'
+      searchButtonVal: 'Search restaurants',
+      fields: {
+        Neighborhood: undefined,
+        Cuisine: undefined
+      }
     };
+    Object.keys(this.state.fields).forEach(this.getFieldValues.bind(this));
```
Comment (stephaniesara) [inline client/components/sidebar.jsx]: I might include this loop within the getFieldValues function instead, and then here in the constructor call the getFieldValues function. Could also rename to populateFieldValues:)

**Pull request closed without merging**